frontend/src/app/components/category-management/category-management.component.ts

```diff
@@ -29,4 +31,22 @@ export class CategoryManagementComponent implements OnInit {
     //   this.data = result;
     // });
   }
+  async saveCategory(
+    name: string,
+    description: string,
+    minHumidityLevel: string
+  ) {
+    this.category.name = name;
+    this.category.description = description;
+    this.category.minHumidityLevel = Number(minHumidityLevel);
+    await this.defaultService.plantTypePost(this.category).subscribe();
+    await new Promise((resolve) => setTimeout(resolve, 300));
+    location.reload();
+  }
+
+  async deleteItem(data: any) {
+    await this.defaultService.plantTypePlantTypeIdDelete(data.id).subscribe();
```

Comment:
If I remember correctly, subscribe() is sync and thus does not need await.
```suggestion
    this.defaultService.plantTypePlantTypeIdDelete(data.id).subscribe();
```

Furthermore, the below code for an artificial timeout can be replaced by using the subscribe() methods callback.

```suggestion
    this.defaultService.plantTypePlantTypeIdDelete(data.id).subscribe({
        next(answer) {
            location.reload();
        },
        error(msg) {
            console.log('Error posting category: ', msg);
       }
    }
););
```